Index: src/main/java/com/tririga/pub/workflow/CustomBusinessConnectTask.java
===================================================================
diff --git a/src/main/java/com/tririga/pub/workflow/CustomBusinessConnectTask.java b/src/main/java/com/tririga/pub/workflow/CustomBusinessConnectTask.java
new file mode 100644
--- /dev/null	
+++ b/src/main/java/com/tririga/pub/workflow/CustomBusinessConnectTask.java	
@@ -0,0 +1,8 @@
+package com.tririga.pub.workflow;
+
+import com.tririga.ws.TririgaWS;
+
+public interface CustomBusinessConnectTask {
+
+    boolean execute(TririgaWS tririgaWS, long userId, Record[] records);
+}
Index: src/main/java/com/tririga/pub/workflow/CustomTask.java
===================================================================
diff --git a/src/main/java/com/tririga/pub/workflow/CustomTask.java b/src/main/java/com/tririga/pub/workflow/CustomTask.java
new file mode 100644
--- /dev/null	
+++ b/src/main/java/com/tririga/pub/workflow/CustomTask.java	
@@ -0,0 +1,7 @@
+package com.tririga.pub.workflow;
+
+import com.tririga.ws.TririgaWS;
+
+public interface CustomTask {
+    boolean execute(Record[] records);
+}
Index: src/main/java/com/tririga/pub/workflow/CustomParamTask.java
===================================================================
diff --git a/src/main/java/com/tririga/pub/workflow/CustomParamTask.java b/src/main/java/com/tririga/pub/workflow/CustomParamTask.java
new file mode 100644
--- /dev/null	
+++ b/src/main/java/com/tririga/pub/workflow/CustomParamTask.java	
@@ -0,0 +1,9 @@
+package com.tririga.pub.workflow;
+
+import java.util.Map;
+
+public interface CustomParamTask {
+
+    CustomParamTaskResult execute(Map var1, long userId, Record[] records);
+
+}
Index: src/main/java/com/tririga/pub/workflow/CustomParamTaskResult.java
===================================================================
diff --git a/src/main/java/com/tririga/pub/workflow/CustomParamTaskResult.java b/src/main/java/com/tririga/pub/workflow/CustomParamTaskResult.java
new file mode 100644
--- /dev/null	
+++ b/src/main/java/com/tririga/pub/workflow/CustomParamTaskResult.java	
@@ -0,0 +1,11 @@
+package com.tririga.pub.workflow;
+
+import java.util.Map;
+
+public interface CustomParamTaskResult {
+
+    boolean getExecutionWasSuccessful();
+
+    Map getReturnParameters();
+}
+
